cou/steps/hypervisor.py

```diff
@@ -160,6 +160,20 @@ def get_azs(self, target: OpenStackRelease) -> AZs:
 
         return azs
 
+    def _upgrade_plan_sanity_checks(self, target: OpenStackRelease) -> None:
+        """Run sanity checks for al applications.
+
+        :param target: OpenStack codename to upgrade.
+        :type target: OpenStackRelease
+        """
+        for app in self.apps:
+            logger.info("running sanity checks for '%s'", app.name)
+            # Note(rgildein): We don't catch the error here because we shouldn't generate any
+            #                 update plan if sanity checks for any application fails.
+            # Note(rgildein): Running sanity checks for all units of applications. We cannot pass
+            #                 None here, since _check_mismatched_versions could be raised.
+            app.upgrade_plan_sanity_checks(target, list(app.units.values()))
```

Comment:
I think we should pass in only in-scope units which are from all AZs groups. Also, maybe we should change the implementation of `_check_mismatched_versions` to also check mismatches when units are passed in since even in hypervisors we should have already filtered out the units that are not in the current release.

Reply:
I agree...

Reply:
So I need to iterate over azs second time to get all apps and all units from each group. I really do not see any benefit of this approach.

I opened the issues #365 for second part of your comment.

Reply:
> At the time the structure was different and was always checking from all units witch could easily raise mismatched versions.

I was asking during time, when we were introducing units to that check.